query/command_test.go

```diff
@@ -70,3 +104,71 @@ func TestCommand_Describe(t *testing.T) {
 		a.EqInt(len(parsedResult), test.length)
 	}
 }
+
+func TestCommand_Select(t *testing.T) {
+	var fakeBackend fakeApiBackend
+	testTimerange, err := api.NewTimerange(0, 120, 30)
+	if err != nil {
+		t.Errorf("Invalid test timerange")
+		return
+	}
+
+	for _, test := range []struct {
+		query       string
+		expectError bool
+		expected    api.SeriesList
+	}{
+		{"select error_series from 0 to 120", true, api.SeriesList{}},
+		{"select series_1 from 0 to 120", false, api.SeriesList{
+			Series: []api.Timeseries{{
+				[]float64{1, 2, 3, 4, 5},
+				api.ParseTagSet("dc=west"),
+			}},
+			Timerange: *testTimerange,
+			Name:      "series_1",
+		}},
+		{"select series_1 + 1 from 0 to 120", false, api.SeriesList{
+			Series: []api.Timeseries{{
+				[]float64{2, 3, 4, 5, 6},
+				api.ParseTagSet("dc=west"),
+			}},
+			Timerange: *testTimerange,
+			Name:      "",
+		}},
+		{"select series_1 * 2 from 0 to 120", false, api.SeriesList{
+			Series: []api.Timeseries{{
+				[]float64{2, 4, 6, 8, 10},
+				api.ParseTagSet("dc=west"),
+			}},
+			Timerange: *testTimerange,
+			Name:      "",
+		}},
+		{"select aggregate.max(series_2) from 0 to 120", false, api.SeriesList{
+			Series: []api.Timeseries{{
+				[]float64{3, 2, 3, 6, 5},
+				api.NewTagSet(),
+			}},
+			Timerange: *testTimerange,
+			Name:      "series_2",
+		}},
+	} {
+		a := assert.New(t).Contextf("query=%s", test.query)
+		rawCommand, err := Parse(test.query)
+		if err != nil {
+			a.Errorf("Unexpected error while parsing")
+			continue
+		}
+		command := rawCommand.(*SelectCommand)
+		rawResult, err := command.Execute(fakeBackend)
+		if err != nil {
+			if !test.expectError {
+				a.Errorf("Unexpected error while executing: %s", err.Error())
+			}
+		} else {
+			casted := rawResult.([]value)
+			first, _ := casted[0].toSeriesList(api.Timerange{})
+			a.Eq(first.Series[0].Values, test.expected.Series[0].Values)
```

Comment:
Should probably have a `.FloatEq()` which allows a small amount of error (like `1e-10`), or it might be hard to test things involving division if they give values which have some numerical error.